### ISSUE TITLE
"Enable public preview" feature to share draft posts

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#21656]
**Is your feature request related to a problem? Please describe.**
The client wanted me to share a preview of an article before I published it.
If I share the draft preview externally they will not have access to it.

**Describe the solution you'd like**
On https://make.wordpress.org/core as part of the notetaker group we share agenda and summary drafts with each other before publishing on make core. It makes it possible for us to go through and recommend adjustments before posting for all to see.

**Describe alternatives you've considered**
I wondered how make core enabled the "Enable public preview" feature and did a search and located @ocean90 Dominik's plugin: https://wordpress.org/plugins/public-post-preview/

It would be very helpful to have the feature included in the Gutenberg publish panel making it possible for anyone to share a preview link to someone not logged into the site.

![Screen Shot 2020-04-16 at 18 41 30](https://user-images.githubusercontent.com/5323259/79483072-092e0900-8012-11ea-9a7d-2780c2bbc66a.png)

**Comment by swissspidy** [comment]
The Public Post Preview plugin is what's used on make/core, FWIW

**Comment by jasmussen** [comment]
I love this feature, and I love that plugin. To me that plugin feels like an excellent way to provide this feature to WordPress and the block editor.

However I'm not sure that as the feature has been built today, that it should be included in the editor by default. It is a somewhat advanced feature, and I suspect that people might inadvertently enable such a preview without really knowing what it means.

If we zoom out a little bit, the use case is not necessarily the ability to preview a post without logging in alone. That's an aspect of it, but if we were to build such a feature directly into the editor today, I'd go the Google Docs model and provide a big beautiful "share" button that would enable users to invite others to not only edit, but also comment. Or just preview. 

In other words, instead of including that plugin, I would move forward with the collaborative editing ticket, #1930, and the annotation API, #3026. Combined, they would solve the flow outlined in this ticket, but arguably in a better way. 

I've added a "needs decision" label in the meantime, to hear whether others think the plugin is worth bringing in, vs. waiting for collaborative editing.

**Comment by paaljoachim** [comment]
Sounds like a very good idea Joen!
In the mean time we can use Dominik's plugin. (As we do on make core)
I have also installed the plugin on a client site and will use it there as long as it is needed.
I look forward to having an even better integrated feature in place in Gutenberg Core.

**Comment by paaljoachim** [comment]
I will go ahead and close this issue as work is going on in other issues.
